src/openassetio-core-c/include/openassetio/c/InfoDictionary.h

```diff
@@ -33,6 +33,14 @@ extern "C" {
 // NOLINTNEXTLINE(modernize-use-using)
 typedef struct OPENASSETIO_NS(InfoDictionary_t) * OPENASSETIO_NS(InfoDictionary_h);
 
+// NOLINTNEXTLINE(modernize-use-using)
+typedef enum {
```

Comment:
[fix] Be good to record the rationale for the choice of an `enum` here over the other options.

Reply:
Added a docstring for this enum in eadd4de7.  Hopefully it answers the question why we're using an enum implicitly, but I can be more explicit if you think its necessary.

Reply:
Thats fine, it might be worth calling out the other options considered in the commit message itself? As a mini DR.

Reply:
Added a note in the commit message in 42b3988c